patches/minecraft/net/minecraft/entity/EntityType.java.patch

```diff
@@ -20,7 +20,90 @@
+       EntityType<T> entitytype = p_200712_1_.func_206830_a(p_200712_0_);
+@@ -244,6 +250,18 @@
+       return field_200787_a.func_82594_a(ResourceLocation.func_208304_a(p_200713_0_));
+    }
+ 
++   public EntityType(Class<? extends T> clazz, Function<? super World, ? extends T> factory, boolean serializable, boolean summonable, @Nullable Type<?> dataFixerType,
++                     boolean useVanillaSpawning, Function<net.minecraftforge.fml.network.FMLPlayMessages.SpawnEntity, Entity> customSpawnCallback,
++                     boolean hasCustomTracking, int range, int updateFreq, boolean sendVelocityUpdates) {
++      this(clazz, factory, serializable, summonable, dataFixerType);
++      this.useVanillaSpawning = useVanillaSpawning;
++      this.customSpawnCallback = customSpawnCallback;
++      this.hasCustomTracking = hasCustomTracking;
++      this.customTrackingRange = range;
++      this.customUpdateFrequency = updateFreq;
++      this.customSendVelocityUpdates = sendVelocityUpdates;
++   }
++
+    public EntityType(Class<? extends T> p_i49579_1_, Function<? super World, ? extends T> p_i49579_2_, boolean p_i49579_3_, boolean p_i49579_4_, @Nullable Type<?> p_i49579_5_) {
+       this.field_201761_aK = p_i49579_1_;
+       this.field_200732_aK = p_i49579_2_;
+@@ -375,15 +393,32 @@
+       return p_200719_1_ == null ? null : p_200719_1_.func_200721_a(p_200719_0_);
+    }
+ 
++   public boolean hasCustomTracking() { return hasCustomTracking; }
++   public int getTrackingRange() { return customTrackingRange; }
++   public int getUpdateFrequency() { return customUpdateFrequency; }
++   public boolean shouldSendVelocityUpdates() { return customSendVelocityUpdates; }
++   public boolean usesVanillaSpawning() { return useVanillaSpawning; }
++   @Nullable public Entity handleSpawnMessage(World world, net.minecraftforge.fml.network.FMLPlayMessages.SpawnEntity msg)
++   {
++      return customSpawnCallback == null ? func_200721_a(world) : customSpawnCallback.apply(msg);
++   }
++
+    public static class Builder<T extends Entity> {
+       private final Class<? extends T> field_201759_a;
+       private final Function<? super World, ? extends T> field_200709_a;
+       private boolean field_200710_b = true;
+       private boolean field_200711_c = true;
++      private boolean useVanillaSpawning;
++      private Function<net.minecraftforge.fml.network.FMLPlayMessages.SpawnEntity, Entity> customSpawnCallback = null;
++      private boolean hasCustomTracking = false;
++      private int trackingRange;
++      private int updateFrequency;
++      private boolean sendVelocityUpdates;
+ 
+       private Builder(Class<? extends T> p_i48756_1_, Function<? super World, ? extends T> p_i48756_2_) {
+          this.field_201759_a = p_i48756_1_;
+          this.field_200709_a = p_i48756_2_;
++         this.useVanillaSpawning = field_201759_a.getName().startsWith("net.minecraft");
```

Comment:
this is weird but since modded and vanilla entities share the same builder now we need a sane default for vanilla.
In 1.12 this was always false in EntityEntryBuilder but that only worked because vanilla entities didn't use EntityEntryBuilder.

Reply:
Use `net.minecraft.` instead.